scalar/string.go

```diff
@@ -78,12 +79,37 @@ func (s *String) Set(val any) error {
 	return nil
 }
 
-type _smallString String
-
 type LargeString struct {
-	_smallString
+	s String
```

Comment:
```suggestion
	String
```
would also work & wouldn't require from you to redefine all of the functions

Reply:
Doesn't work because we have `.String()` from fmt.Stringer